tests/unit/amo/pages/TestUserProfile.js

```diff
@@ -619,209 +567,214 @@ describe(__filename, () => {
     const userId = 1;
-    const { store } = signInUserWithProps({
+    signInUserWithProps({
       userId,
       permissions: [USERS_EDIT],
     });
 
     const user = createUserAccountResponse({ userId });
     store.dispatch(loadUserAccount({ user }));
 
-    const root = renderUserProfile({ params: { userId }, store });
+    renderUserProfile({ params: { userId } });
 
-    expect(root.find('.UserProfile-admin-link')).toHaveLength(1);
-    expect(root.find('.UserProfile-admin-link')).toHaveProp(
+    const adminButton = screen.getByRole('link', { name: 'Admin user' });
+    expect(adminButton).toBeInTheDocument();
+    expect(screen.getByRole('link', { name: 'Admin user' })).toHaveAttribute(
       'href',
       `/admin/models/users/userprofile/${userId}/`,
     );
   });
 
   it('does not render an admin link if user is not allowed to admin users', () => {
     const userId = 1;
-    const { store } = signInUserWithProps({
+    signInUserWithProps({
       userId,
       permissions: [],
     });
 
     const user = createUserAccountResponse({ userId });
     store.dispatch(loadUserAccount({ user }));
 
-    const root = renderUserProfile({ params: { userId }, store });
+    renderUserProfile({ params: { userId } });
 
-    expect(root.find('.UserProfile-admin-link')).toHaveLength(0);
+    expect(
+      screen.queryByRole('link', { name: 'Admin user' }),
+    ).not.toBeInTheDocument();
   });
 
   it('does not dispatch any user actions when there is an error', () => {
-    const { store } = dispatchClientMetadata();
-    const fakeDispatch = sinon.spy(store, 'dispatch');
+    const dispatch = jest.spyOn(store, 'dispatch');
 
-    const errorHandler = new ErrorHandler({
-      id: 'some-id',
-      dispatch: fakeDispatch,
+    const userId = 123;
+    createFailedErrorHandler({
+      error: new Error(),
+      id: createErrorHandlerId({ userId }),
+      message: 'unexpected error',
+      store,
     });
-    errorHandler.handle(new Error('unexpected error'));
-
-    fakeDispatch.resetHistory();
 
-    renderUserProfile({ errorHandler, store });
+    renderUserProfile({ params: { userId } });
 
-    sinon.assert.neverCalledWithMatch(fakeDispatch, {
-      type: FETCH_USER_ACCOUNT,
-    });
-    sinon.assert.neverCalledWithMatch(fakeDispatch, { type: FETCH_REVIEWS });
+    expect(dispatch).not.toHaveBeenCalledWith(
+      expect.objectContaining({ 'type': FETCH_USER_ACCOUNT }),
+    );
+    expect(dispatch).not.toHaveBeenCalledWith(
+      expect.objectContaining({ 'type': FETCH_REVIEWS }),
+    );
+    expect(dispatch).not.toHaveBeenCalledWith(
+      expect.objectContaining({ 'type': FETCH_USER_REVIEWS }),
+    );
   });
 
   it('fetches reviews if not loaded and userId does not change', () => {
-    const { params, store } = signInUserWithUserId(100);
+    const userId = 100;
+    const { params } = signInUserWithProps({ userId });
     const user = getCurrentUser(store.getState().users);
```

Comment:
I don't think we need this. It's only ever used for `user.id`, below, but that is going to be the same thing as the `userId` that you defined above.